### DISCUSSION TITLE
getting error

### DISCUSSION THREAD

**Discussion by shubhamjain108-lab** [chatwoot/chatwoot#2437] [Q&A] [answered]
When i am executing command  sudo docker-compose run --rm rails bundle exec rails db:reset

getting error :

Starting chatwoot_redis_1    ... done
Starting chatwoot_mailhog_1  ... done
Starting chatwoot_webpack_1  ... done
Starting chatwoot_mailhog_1 ...
Starting chatwoot_postgres_1 ... done
Starting chatwoot_sidekiq_1  ... done
+ rm -rf /app/tmp/pids/server.pid
+ rm -rf '/app/tmp/cache/*'
+ echo 'Waiting for postgres to become ready....'
Waiting for postgres to become ready....
+ docker/entrypoints/helpers/pg_database_url.sh
+ export 'POSTGRES_PORT=5532'
+ PG_READY='pg_isready -h postgres -p 5532 -U postgres'
+ pg_isready -h postgres -p 5532 -U postgres
postgres:5532 - no response
+ sleep 2
+ pg_isready -h postgres -p 5532 -U postgres
postgres:5532 - no response
+ sleep 2
+ pg_isready -h postgres -p 5532 -U postgres
postgres:5532 - no response
+ sleep 2
+ pg_isready -h postgres -p 5532 -U postgres
postgres:5532 - no response
+ sleep 2
+ pg_isready -h postgres -p 5532 -U postgres
postgres:5532 - no response

**Comment by sojan-official** [comment]
@shubhamjain108-lab could you specify the docs which you followed for set up ?

**Reply by shubhamjain108-lab** [reply]
https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker

in this step 4) docker-compose run --rm rails bundle exec rails db:reset

error is getting .

**Reply by shubhamjain108-lab** [reply]
getting error on sudo docker-compose up

vagrant@app:~/chatwoot$ sudo docker-compose up
Starting chatwoot_mailhog_1  ... done
Starting chatwoot_base_1     ... done
Starting chatwoot_webpack_1  ... done
Starting chatwoot_postgres_1 ... done
Starting chatwoot_redis_1    ... done
Starting chatwoot_sidekiq_1  ... done
Starting chatwoot_rails_1    ... done
Attaching to chatwoot_webpack_1, chatwoot_postgres_1, chatwoot_base_1, chatwoot_mailhog_1, chatwoot_redis_1, chatwoot_sidekiq_1, chatwoot_rails_1
mailhog_1   | 2021/06/16 11:40:37 Using in-memory storage
mailhog_1   | 2021/06/16 11:40:37 [SMTP] Binding to address: 0.0.0.0:1025
mailhog_1   | 2021/06/16 11:40:37 Serving under http://0.0.0.0:8025/
mailhog_1   | [HTTP] Binding to address: 0.0.0.0:8025
mailhog_1   | Creating API v1 with WebPath:
mailhog_1   | Creating API v2 with WebPath:
postgres_1  |
postgres_1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1  |
postgres_1  | 2021-06-16 11:40:38.005 UTC [1] LOG:  starting PostgreSQL 12.7 (Debian 12.7-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
postgres_1  | 2021-06-16 11:40:38.006 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2021-06-16 11:40:38.006 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2021-06-16 11:40:38.059 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2021-06-16 11:40:38.253 UTC [24] LOG:  database system was shut down at 2021-06-16 11:37:45 UTC
postgres_1  | 2021-06-16 11:40:38.328 UTC [1] LOG:  database system is ready to accept connections
rails_1     | + rm -rf /app/tmp/pids/server.pid
rails_1     | + rm -rf '/app/tmp/cache/*'
rails_1     | + echo 'Waiting for postgres to become ready....'
rails_1     | Waiting for postgres to become ready....
rails_1     | + docker/entrypoints/helpers/pg_database_url.sh
redis_1     | 1:C 16 Jun 2021 11:40:36.505 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis_1     | 1:C 16 Jun 2021 11:40:36.506 # Redis version=6.2.4, bits=64, commit=00000000, modified=0, pid=1, just started
redis_1     | 1:C 16 Jun 2021 11:40:36.506 # Configuration loaded
redis_1     | 1:M 16 Jun 2021 11:40:36.507 * monotonic clock: POSIX clock_gettime
redis_1     | 1:M 16 Jun 2021 11:40:36.573 * Running mode=standalone, port=6379.
redis_1     | 1:M 16 Jun 2021 11:40:36.573 # Server initialized
redis_1     | 1:M 16 Jun 2021 11:40:36.573 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis_1     | 1:M 16 Jun 2021 11:40:36.636 * Loading RDB produced by version 6.2.4
redis_1     | 1:M 16 Jun 2021 11:40:36.636 * RDB age 173 seconds
redis_1     | 1:M 16 Jun 2021 11:40:36.636 * RDB memory usage when created 0.77 Mb
redis_1     | 1:M 16 Jun 2021 11:40:36.636 * DB loaded from disk: 0.063 seconds
redis_1     | 1:M 16 Jun 2021 11:40:36.636 * Ready to accept connections
rails_1     | + export 'POSTGRES_PORT=5532'
rails_1     | + PG_READY='pg_isready -h localhost -p 5532 -U postgres'
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
yarn install v1.22.10
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
[1/5] Validating package.json...
[2/5] Resolving packages...
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
sidekiq_1   | DEPRECATION WARNING: Devise::Models::Authenticatable::BLACKLIST_FOR_SERIALIZATION is deprecated! Use Devise::Models::Authenticatable::UNSAFE_ATTRIBUTES_FOR_SERIALIZATION instead. (called from block (2 levels) in require at /usr/local/bundle/gems/bundler-2.2.20/lib/bundler/runtime.rb:66)
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
[3/5] Fetching packages...
error Couldn't find the binary git
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
mailhog_1   | [APIv1] KEEPALIVE /api/v1/events
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2
rails_1     | + pg_isready -h localhost -p 5532 -U postgres
rails_1     | localhost:5532 - no response
rails_1     | + sleep 2

**Comment by sojan-official** [comment]
Please reset the older images and try with the updated instructions 
https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker

mainly you will have to set values for REDIS_PASSWORD and POSTGRES_PASSWORD in your `.env` file 
and also update `docker-compose.yml` with the same  POSTGRES_PASSWORD

**Comment by shubhamjain108-lab** [comment]
same issue is coming,


vagrant@app:~/chatwoot$ sudo  docker-compose run --rm rails bundle exec rails db:chatwoot_prepare
Starting chatwoot_mailhog_1 ... done
Starting chatwoot_webpack_1  ... done
Starting chatwoot_postgres_1 ... done
Starting chatwoot_redis_1    ... done
Starting chatwoot_sidekiq_1  ... done
+ rm -rf /app/tmp/pids/server.pid
+ rm -rf '/app/tmp/cache/*'
+ echo 'Waiting for postgres to become ready....'
Waiting for postgres to become ready....
+ docker/entrypoints/helpers/pg_database_url.sh
+ export 'POSTGRES_PORT=5532'
+ PG_READY='pg_isready -h localhost -p 5532 -U postgres'
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres
localhost:5532 - no response
+ sleep 2
+ pg_isready -h localhost -p 5532 -U postgres

**Reply by sojan-official** [reply]
@shubhamjain108-lab  did you delete the previous images built before updating the POSTGRES_PASSWORD in both `.env` file and `docker-compose.yml` ?

**Reply by shubhamjain108-lab** [reply]
After pull new image for postgresql it is same error,



vagrant@app:~/chatwoot$ sudo docker-compose run --rm rails bundle exec rails db:chatwoot_prepare
Pulling postgres (postgres:12)...
12: Pulling from library/postgres
69692152171a: Already exists
a31b993d5cc6: Pull complete
f65921886500: Pull complete
b9c1a94e4ca8: Pull complete
435dd99ceb68: Pull complete
d3ee8e88c67c: Pull complete
84b08674f942: Pull complete
7d358e850d3e: Pull complete
b22f4aed6cbf: Pull complete
59a545658c54: Pull complete
3a8bab9649ea: Pull complete
299c2339be6d: Pull complete
1c358064ba1f: Pull complete
Digest: sha256:097e6fc733e01597390d5190a6b99744e75da91c1bce628b2dedea3acd68958b
Status: Downloaded newer image for postgres:12
Starting chatwoot_webpack_1  ... done
Creating chatwoot_postgres_1 ... done
Starting chatwoot_sidekiq_1  ... done
Creating chatwoot_rails_run  ... done
+ rm -rf /app/tmp/pids/server.pid
+ rm -rf '/app/tmp/cache/*'
+ echo 'Waiting for postgres to become ready....'
Waiting for postgres to become ready....
+ docker/entrypoints/helpers/pg_database_url.sh
+ export 'POSTGRES_PORT=5432'
+ PG_READY='pg_isready -h localhost -p 5432 -U postgres'
+ pg_isready -h localhost -p 5432 -U postgres
localhost:5432 - no response
+ sleep 2
+ pg_isready -h localhost -p 5432 -U postgres
localhost:5432 - no response
+ sleep 2
+ pg_isready -h localhost -p 5432 -U postgres
localhost:5432 - no response

**Reply by shubhamjain108-lab** [reply]
could you please share any other way to setup chatwoot in development, as docker is not working?

**Reply by sojan-official** [reply]
- head over to https://www.chatwoot.com/docs/contributing-guide
- from the sidebar, chose instructions for the operating system you are using
- finish the environment setup 
- then follow the setup guide

Chatwoot is a standard rails application. So if you face any error in running, you could also refer to the StackOverflow answers around it.

**Reply by shubhamjain108-lab** [reply]
12:04:55 backend.1  | ActionView::Template::Error (Webpacker can't find superadmin.js in /home/vagrant/chatwoot/public/packs/manifest.json. Possible causes:
12:04:55 backend.1  | 1. You want to set webpacker.yml value of compile to true for your environment
12:04:55 backend.1  |    unless you are using the `webpack -w` or the webpack-dev-server.
12:04:55 backend.1  | 2. webpack has not yet re-run to reflect updates.
12:04:55 backend.1  | 3. You have misconfigured Webpacker's config/webpacker.yml file.
12:04:55 backend.1  | 4. Your webpack configuration is not creating a manifest.
12:04:55 backend.1  | Your manifest contains:
12:04:55 backend.1  | {
12:04:55 backend.1  | }
12:04:55 backend.1  | ):
12:04:55 backend.1  |     2: <html>
12:04:55 backend.1  |     3:   <head>
12:04:55 backend.1  |     4:     <title>SuperAdmin | Chatwoot</title>
12:04:55 backend.1  |     5:     <%= javascript_pack_tag 'superadmin' %>
12:04:55 backend.1  |     6:     <%= stylesheet_pack_tag 'superadmin' %>
12:04:55 backend.1  |     7:   </head>
12:04:55 backend.1  |     8:   <body data-gr-c-s-loaded="true">
12:04:55 backend.1  |
12:04:55 backend.1  | app/views/installation/onboarding/index.html.erb:5
12:04:55 backend.1  | app/controllers/application_controller.rb:18:in `handle_with_exception'
12:04:55 backend.1  | app/controllers/concerns/switch_locale.rb:24:in `set_locale'
12:04:55 backend.1  | app/controllers/concerns/switch_locale.rb:11:in `switch_locale'
12:04:56 worker.1   | 2021-06-19T12:04:56.970Z pid=22814 tid=nie INFO: Booting Sidekiq 6.1.1 with redis options {:url=>"redis://localhost:6379", :password=>"REDACTED", :reconnect_attempts=>2, :network_timeout=>5, :size=>25}

**Comment by shubhamjain108-lab** [comment, marked as answer]
this point is working head over to https://www.chatwoot.com/docs/contributing-guide, it is working, Thank you sojan-official
